server/content-repo/content-delivery.router.js

```diff
@@ -0,0 +1,40 @@
+const { ContentRepo, Enrollment } = require('../common/database');
+const { createError } = require('../common/errors');
+const ctrl = require('./content-delivery.controller');
+const HttpStatus = require('http-status');
+const router = require('express').Router();
+
+const { NOT_FOUND, UNAUTHORIZED } = HttpStatus;
+
+router
+  .use('/*', hasAccess)
+  .get('/', ctrl.list)
+  .use('/:contentId*', getRepo)
+  .get('/:contentId', ctrl.get)
+  .get('/:contentId/container/:containerId', ctrl.getContainer)
+  .get('/:contentId/exam/:examId', ctrl.getExam);
+
+function hasAccess({ cohort, user }, res, next) {
+  if (user.isAdmin()) return next();
+  const opts = { where: { studentId: user.id, cohortId: cohort.id } };
+  return Enrollment.count(opts)
```

Comment:
Count will result with table scan. You just need to find one (Since there should be only one).